pydantic/main.py

```diff
@@ -674,23 +668,23 @@ def _iter(
                     exclude_defaults=exclude_defaults,
                     exclude_none=exclude_none,
                 )
-                if not (exclude_none and value is None):
-                    yield k, value
+            yield k, v
 
     def _calculate_keys(
         self,
         include: Optional[Union['AbstractSetIntStr', 'DictIntStrAny']],
         exclude: Optional[Union['AbstractSetIntStr', 'DictIntStrAny']],
         exclude_unset: bool,
         update: Optional['DictStrAny'] = None,
-    ) -> Optional['SetStr']:
+    ) -> Optional[Union['SetStr', KeysView[str]]]:
```

Comment:
```suggestion
    ) -> Optional[AbstractSet[str]]:
```
?

Reply:
Looks great. Should I change it in other places?

Reply:
Yes please, if in this file.